File: internal/adapters/guest/fixtures/jvm-java-spring/src/main/java/dev/mds/Application.java
package dev.mds;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

@SpringBootApplication
public class Application {
    public static void main(String[] args) throws Exception {
        ConfigurableApplicationContext context = SpringApplication.run(Application.class, args);
        if (!"1".equals(System.getenv("MDS_CAPABILITY_PROBE"))) {
            return;
        }
        int port = Integer.parseInt(context.getEnvironment().getRequiredProperty("local.server.port"));
        var request = HttpRequest.newBuilder(URI.create("http://127.0.0.1:" + port + "/greeting")).build();
        var response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
        if (response.statusCode() != 200 || !response.body().contains("hello-java")) {
            throw new IllegalStateException("Spring Boot Java probe endpoint failed");
        }
        SpringApplication.exit(context);
    }
}

File: internal/adapters/guest/fixtures/jvm-kotlin-spring/src/main/kotlin/dev/mds/Application.kt
package dev.mds

import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class Application

fun main(args: Array<String>) {
    val context = runApplication<Application>(*args)
    if (System.getenv("MDS_CAPABILITY_PROBE") != "1") return
    val port = context.environment.getRequiredProperty("local.server.port")
    val request = HttpRequest.newBuilder(URI.create("http://127.0.0.1:$port/greeting")).build()
    val response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString())
    check(response.statusCode() == 200 && response.body().contains("hello-kotlin")) {
        "Spring Boot Kotlin probe endpoint failed"
    }
    context.close()
}
